cmake/External/ucc.cmake

```diff
@@ -0,0 +1,20 @@
+if(NOT __UCC_INCLUDED)
+  set(__UCC_INCLUDED TRUE)
+
+  if(USE_SYSTEM_UCC)
+    set(UCX_HOME $ENV{UCX_HOME} CACHE PATH "UCX install directory")
+    set(UCC_HOME $ENV{UCC_HOME} CACHE PATH "UCC install directory")
```

Comment:
Should we use [`find_package`](https://cmake.org/cmake/help/latest/command/find_package.html) to find the path for UCC and UCX? I added cmake support to UCX and UCC in the past:
https://github.com/openucx/ucx/pull/7096
https://github.com/openucx/ucc/pull/255

Reply:
This PR only supports USE_SYSTEM_UCC for simplicity. I have a follow up PR on making UCC/UCX repos part of pytorch/third-party (and automatically compile them with pytorch cmake etc). I feel those are very cmake-specific and can be addressed as a follow up so that this PR can focus on UCC PG (which is already pretty big PR). I will look at your past PRs accordingly. Thanks for the pointers!

Reply:
sgtm. thanks.